gobblin-core/src/main/java/org/apache/gobblin/writer/PartitionedDataWriter.java

```diff
@@ -170,9 +175,12 @@ public DataWriter<D> get() {
                     try {
                       log.info(String.format("Adding one more writer to loading cache of existing writer "
                           + "with size = %d", partitionWriters.size()));
-                      return createPartitionWriter(key);
-                    } catch (IOException e) {
+                      Future<DataWriter<D>> future = createWriterPool.submit(() -> createPartitionWriter(key));
+                      return future.get(writeTimeoutInterval, TimeUnit.SECONDS);
```

Comment:
The builder shares the timeout interval with writeTimeoutInterval. I saw you made a couple commit tuning the default writeTimeoutInterval. Will that be a concern if we explicitly set it?

Reply:
We won't need any config change here. If the job originally has a timeout for writing one record, we will enforce the same timeout for creating one writer, if the job does not have that timeout, we will have no timeout for creating a writer as well.

Reply:
We won't need any config change here. If the job originally has a timeout for writing one record, we will enforce the same timeout for creating one writer, if the job does not have that timeout, we will have no timeout for creating a writer as well.